62.CAD/common.hlsl

```diff
@@ -22,10 +24,53 @@ struct DrawObject
     uint64_t geometryAddress;
 };
 
+template <typename float_t>
+struct QuadBezierAnalyticArcLengthCalculator
+{
+#ifdef __cplusplus
+    using float2_t = nbl::core::vector2d<float_t>;
+#else
+    using float2_t = vector<float_t, 2>;
+#endif
+
+    static QuadBezierAnalyticArcLengthCalculator construct(float_t lenA2, float_t AdotB, float_t a, float_t b, float_t  c, float_t b_over_4a)
+    {
+        QuadBezierAnalyticArcLengthCalculator ret = { lenA2, AdotB, a, b, c, b_over_4a };
+        return ret;
+    }
+    
+    static QuadBezierAnalyticArcLengthCalculator construct(float2_t P0, float2_t P1, float2_t P2)
+    {
+        QuadBezierAnalyticArcLengthCalculator ret;
+        float2_t A = P0 - 2.0 * P1 + P2;
+        float2_t B = 2.0 * (P1 - P0);
+        float2_t C = P0;
+        ret.lenA2 = dot(A, A);
+        ret.AdotB = dot(A, B);
+
+        ret.a = 4.0 * ret.lenA2;
+        ret.b = 4.0 * ret.AdotB;
+        ret.c = dot(B, B);
+
+        ret.b_over_4a = ret.AdotB / ret.a;
+        return ret;
+    }
+
+    float_t lenA2;
+    float_t AdotB;
+
+    float_t a;
+    float_t b;
+    float_t c;
+
+    float_t b_over_4a;
+};
+
 struct QuadraticBezierInfo
 {
     double2 p[3]; // 16*3=48bytes
     // TODO[Przemek]: Any Data related to precomputing things for beziers will be here
```

Comment:
remove this TODO and all the other TODOs I had previously put for you that are resovled

Reply:
done